File: CS/HowToAccessApplicationModelInCode.Module.Win/SwitchController.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.Model;

namespace HowToAccessApplicationModelInCode.Module.Win {
    public partial class SwitchController : ViewController {
        public SwitchController() {
            InitializeComponent();
            RegisterActions(components);
        }
        private void SwitchAction_Execute(object sender, SimpleActionExecuteEventArgs e) {
            if (((IModelListView)View.Model).MasterDetailMode == 
                MasterDetailMode.ListViewAndDetailView)
                ((IModelListView)View.Model).MasterDetailMode = 
                    DevExpress.ExpressApp.MasterDetailMode.ListViewOnly;
            else
                ((IModelListView)View.Model).MasterDetailMode = 
                    DevExpress.ExpressApp.MasterDetailMode.ListViewAndDetailView;
            View.SetModel(View.Model);
        }
    }
}

File: VB/HowToAccessApplicationModelInCode.Module.Win/SwitchController.vb
Imports System
Imports System.ComponentModel
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Text

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base
Imports DevExpress.ExpressApp.Model

Namespace HowToAccessApplicationModelInCode.Module.Win
    Partial Public Class SwitchController
        Inherits ViewController

        Public Sub New()
            InitializeComponent()
            RegisterActions(components)
        End Sub
       Private Sub SwitchAction_Execute(ByVal sender As Object, ByVal e As SimpleActionExecuteEventArgs) Handles SwitchAction.Execute
            Dim model As IModelListView = DirectCast(View, ListView).Model
            Dim shortcut As ViewShortcut = Frame.View.CreateShortcut()
            Frame.SetView(Nothing)
            If model.MasterDetailMode = MasterDetailMode.ListViewAndDetailView Then
                model.MasterDetailMode = MasterDetailMode.ListViewOnly
            Else
                model.MasterDetailMode = MasterDetailMode.ListViewAndDetailView
            End If
            Frame.SetView(Application.ProcessShortcut(shortcut), True, Frame)
       End Sub
    End Class
End Namespace
